lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1462,6 +1462,23 @@ static void checkEnumRawValues(TypeChecker &TC, EnumDecl *ED) {
     KnownProtocolKind::ExpressibleByExtendedGraphemeClusterLiteral,
   };
 
+  // The raw type must conform to Equatable, unless we explicitly have a
+  // rawValue and init(rawValue:).
+  if (!conformsToProtocol(KnownProtocolKind::Equatable)) {
+    auto &ctx = TC.Context;
+    auto rawValueInit =
+        DeclName(ctx, DeclBaseName::createConstructor(), {ctx.Id_rawValue});
+    auto doesNotHaveRawValue = ED->lookupDirect(ctx.Id_rawValue).empty();
+    auto doesNotHaveRawValueInit = ED->lookupDirect(rawValueInit).empty();
```

Comment:
`lookupDirect` isn't necessarily correct anyway, since that won't look into extensions. If you go through the normal qualified lookup path maybe we'll get the synthesis we need anyway.

Reply:
Hmm I thought it did when I looked at the implementation, but alright, I'll do it the other way!